src/db/collections/application_state.rs

```diff
@@ -15,7 +15,8 @@ use crate::{
 /// The MongoDb document representation of singleton Application State.
 #[derive(Clone, Debug, Eq, PartialEq, Serialize, Deserialize)]
 pub struct ApplicationStateDocument {
-    pub starting_index: MilestoneIndexTimestamp,
+    pub starting_index: Option<MilestoneIndexTimestamp>,
+    pub last_migration: Option<String>,
```

Comment:
I think this should be a number instead of a string. Also would have the benefit that it's easy too determine how much versions the db is behind. Also the current design doesn't allow two migrations at the same day for now real reason.